weave/scorers/trust_scorer.py

```diff
@@ -0,0 +1,376 @@
+
+    Example:
+        ```python
+        scorer = TrustScorer(run_in_parallel=True)
+
+        # Basic scoring
+        result = scorer.score(
+            output="The sky is blue.",
+            context="Facts about the sky.",
+            query="What color is the sky?"
+        )
+
+        # Example output:
+        WeaveScorerResult(
+            passed=True,
+            metadata={
+                "trust_level": "high_no-issues-found",
+                "critical_issues": [],
+                "advisory_issues": [],
+                "raw_outputs": {
+                    "WeaveToxicityScorerV1": {"passed": True, "extras": {"Race/Origin": 0, "Gender/Sex": 0, "Religion": 0, "Ability": 0, "Violence": 0}},
+                    "WeaveHallucinationScorerV1": {"passed": True, "extras": {"score": 0.1}},
+                    "WeaveContextRelevanceScorerV1": {"passed": True, "extras": {"score": 0.85}},
+                    "WeaveFluencyScorerV1": {"passed": True, "extras": {"score": 0.95}},
+                    "WeaveCoherenceScorerV1": {"passed": True, "extras": {"coherence_label": "Perfectly Coherent", "coherence_id": 4, "score": 0.9}}
+                },
+                "scores": {
+                    "WeaveToxicityScorerV1": {"Race/Origin": 0, "Gender/Sex": 0, "Religion": 0, "Ability": 0, "Violence": 0},
+                    "WeaveHallucinationScorerV1": 0.1,
+                    "WeaveContextRelevanceScorerV1": 0.85,
+                    "WeaveFluencyScorerV1": 0.95,
+                    "WeaveCoherenceScorerV1": 0.9
+                }
+            }
+        )
+        ```
+
+    """
+
+    # Model configuration
+    device: str = Field(
+        default="cpu",
+        description="Device for model inference ('cpu', 'cuda', 'mps', 'auto')",
+        from_default=True,
+    )
+    context_relevance_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the context relevance model",
+        validate_default=True,
+    )
+    hallucination_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the hallucination model",
+        validate_default=True,
+    )
+    toxicity_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the toxicity model",
+        validate_default=True,
+    )
+    fluency_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the fluency model",
+        validate_default=True,
+    )
+    coherence_model_name_or_path: str = Field(
+        default="",
+        description="Path or name of the coherence model",
+        validate_default=True,
+    )
+    run_in_parallel: bool = Field(
+        default=True,
+        description="Whether to run scorers in parallel or sequentially, useful for debugging.",
+    )
+
+    # Define scorer categories
+    _critical_scorers: set[type[weave.Scorer]] = {
```

Comment:
I'm surprised this works.  Doesn't it have to be a private attr?

Reply:
yeap, me to